pkg/controller/atlasproject/backupcompliancepolicy.go

```diff
@@ -0,0 +1,156 @@
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api"
+	akov2 "github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/api/v1"
+	"github.com/mongodb/mongodb-atlas-kubernetes/v2/pkg/controller/workflow"
+)
+
+const bcpNotMet = "BACKUP_POLICIES_NOT_MEETING_BACKUP_COMPLIANCE_POLICY_REQUIREMENTS"
+
+type backupComplianceController struct {
+	ctx     *workflow.Context
+	client  client.Client
+	project *akov2.AtlasProject
+}
+
+func (r *AtlasProjectReconciler) ensureBackupCompliance(ctx *workflow.Context, project *akov2.AtlasProject) workflow.Result {
+	ctx.Log.Debug("starting backup compliance policy processing")
+	defer ctx.Log.Debug("finished backup compliance policy processing")
+
+	b := backupComplianceController{
+		ctx:     ctx,
+		client:  r.Client,
+		project: project,
+	}
+
+	return b.handlePending()
+}
+
+func (b *backupComplianceController) handlePending() workflow.Result {
+	bcp, found, err := b.getAtlasBackupCompliancePolicy()
+	if err != nil {
+		return b.terminate(workflow.Internal, err)
+	}
+
+	akoEmpty := true
+	if bcpRef := b.project.Spec.BackupCompliancePolicyRef; bcpRef != nil {
+		akoEmpty = bcpRef.IsEmpty()
+	}
+
+	atlasEmpty := !found
+
+	switch {
+	case !akoEmpty && atlasEmpty:
+		return b.upsert(bcp)
+	case !akoEmpty && !atlasEmpty:
+		return b.upsert(bcp)
+	case akoEmpty && !atlasEmpty:
+		return b.delete()
+	default:
+		return b.unmanage()
+	}
+}
+
+// upsert updates the backup compliance settings for a project. These settings can only be updated (not created), so
+// we also use this for creation too.
+func (b *backupComplianceController) upsert(atlasBCP *admin.DataProtectionSettings20231001) workflow.Result {
+	b.ctx.Log.Debug("updating backup compliance policy")
+	akoBCP, err := b.getAKOBackupCompliancePolicy()
+	if err != nil {
+		return b.terminate(workflow.Internal, err)
+	}
+	equal, err := cmp.SemanticEqual(akoBCP, akov2.NewBCPFromAtlas(atlasBCP))
+	if err != nil {
+		return b.terminate(workflow.Internal, err)
+	}
+	if !equal {
+		_, _, err = b.ctx.SdkClient.CloudBackupsApi.UpdateDataProtectionSettings(b.ctx.Context, b.project.ID(), akoBCP.ToAtlas(b.project.ID())).OverwriteBackupPolicies(akoBCP.Spec.OverwriteBackupPolicies).Execute()
+		if err != nil {
+			if admin.IsErrorCode(err, bcpNotMet) {
+				return b.terminate(workflow.ProjectBackupCompliancePolicyNotMet, err)
+			}
+			return b.terminate(workflow.ProjectBackupCompliancePolicyNotCreatedInAtlas, err)
+		}
+	}
+	return b.idle()
+}
+
+// delete begins the deletion process for a backup compliance policy. However, the admin API cannot delete BCPs (which must be
+// done via support), so we notify the user of next steps via status.
+func (b *backupComplianceController) delete() workflow.Result {
+	b.ctx.Log.Debug("deleting backup compliance policy")
+	return b.terminate(
+		workflow.ProjectBackupCompliancePolicyCannotDelete,
+		errors.New("cannot delete backup compliance policy through the API - to delete a backup compliance policy, please contact support"),
```

Comment:
```suggestion
		errors.New("deletion of backup compliance policies is not supported in Atlas: please contact support"),
```

Reply:
Linter doesnt like block capital variables like this. Also I remembered we have a file for these Atlas API errors, so added it there instead.